index.html

```diff
@@ -2326,6 +2327,102 @@ <h2>Definition of Roles</h2>
 				</tbody>
 			</table>
 		</div>
+		<div class="role" id="comment">
+			<rdef>comment</rdef>
+			<div class="role-description">
+				<p>A comment contains a remark expressing a reaction to other content within the page.</p>
+				<p>The relationship of comments to other content SHOULD be exposed as follows:</p>
+				<ol>
+					<li>If the comment refers to other content within the page, the source content SHOULD use <pref>aria-details</pref> to point to the related comment.</li>
```

Comment:
Regarding the "many comments for a blog" scenario, why we would prefer to have `aria-details` point to a comment-container element rather than having it point to the list of root comments?
- Is it because it simplifies the code for authors (they wouldn't have to come up with a bunch of unique ids)?
- Or is it possible that the list of ids could get too long for UA to maintain?
- Or are we trying to encourage authors to use a landmark in this use case?

Also (sorry to bring this up if it is already decided) if we are suggesting pointing to a container for the blog comments use case, why aren't we suggesting pointing to a container (even if just a div) for every use case that has more than one comment? (in which case we could consider going back to aria-details having only one id)?

Anyhow, if we allow choosing between a list of ids or a container, the we need to add some words that explain _why_ to use one method or the other, to help authors decide which way to code their particular use case.

Reply:
> Regarding the "many comments for a blog" scenario, why we would prefer to have `aria-details` point to a comment-container element rather than having it point to the list of root comments?

The language is meant to be neutral and not suggest that  there is preference. I think we might work that out in practices.

> * Is it because it simplifies the code for authors (they wouldn't have to come up with a bunch of unique ids)?

Simplify code yes -- Modifying the article attributes  every time a new comment is posted could be problematic in some applications.

> * Or is it possible that the list of ids could get too long for UA to maintain?

That is a good question. I have no idea what the limits are there. We would certainly want to consider this question when writing practices.

> * Or are we trying to encourage authors to use a landmark in this use case?

Not necessarily strong encouragement, but certainly suggesting it for consideration.

> Also (sorry to bring this up if it is already decided) if we are suggesting pointing to a container for the blog comments use case, why aren't we suggesting pointing to a container (even if just a div) for every use case that has more than one comment? (in which case we could consider going back to aria-details having only one id)?

There are certainly going to be designs where multiple comments or other types of annotations for an element will not all be together in the same container. So, we definitely need to support multiple IDs for aria-details.

If there are multiple comments but all in a single thread, there is no need for a container. In that case, only the root comment is referenced by aria-details. However, if the thread is paged, I am not yet clear on how that would work.

> Anyhow, if we allow choosing between a list of ids or a container, the we need to add some words that explain _why_ to use one method or the other, to help authors decide which way to code their particular use case.

I don't think we need that in the spec.

I'm still thinking about the container case. At one time, @aleventhal had proposed a commentsection role. I didn't and still do not think that is the approach that best addresses both the needs of users and AT devs. I am especially not a fan of yet one more role that has such a special purpose semantic. The more roles of that nature we have, the less useful they become to users. Sometimes you want the comment collection in a landmark region, and sometimes you do not. So, making commentsection a region would be a bad idea because it could force region proliferation into some apps. On the other hand, if it is its own special thing, that means locating it is yet one more very specialized function, of which we already have too many by at least one order of magnitude, and approaching 2.

There is definite advantage to knowing what is at the other end of a details relationship. That enables screen readers in particular to provide the user with information the lets them know whether or not the details relationship is something they need to exercise.

Perhaps one way would be to have a property called aria-containsannotations that would be supported on group, list, region, complementary, table, grid, and treegrid. The property could support tokens like comment, suggestion, endnote. Then, if aria-details points to that container, the screen reader would know to look inside and maybe even count the number of elements with each of the types suggested by the value of the property. Now, the screen reader gets to know what to tell the user. The engineer only needs one ID relationship, and the designer of the accessible experience can decide what the best type of container is for the annotations.

Reply:
@carmacleod, we're trying to keep things simple in the common case, but make difficult/rare things possible. It was a compromise but we decided that by having only "comment", it would be simpler and cleaner for most cases. It didn't seem like comment section was worth having.

Reply:
@mcking65  I'm trying some things with your wording to IMO make it simpler. For one thing, I think just saying one case is for replies, and another is for non replies makes it easier to follow, and feels less convoluted, so I'm putting the emphasis more on that.

Reply:
I came back to this issue to strike out my mention of aria-details always pointing to a single container (perfectly happy to let that drop ;) and I saw that you replied already. Thanks for the quick response!

> There are certainly going to be designs where multiple comments or other types of annotations for an element will not all be together in the same container. So, we definitely need to support multiple IDs for aria-details.

I had just come to the same conclusion after re-reading aria-details. The detail elements could conceivably be in multiple places. So it doesn't make sense to require authors to always use a container.

I also realized that comments aren't the only type of annotation that could have more than one instance. Content could easily have footnotes, a term could have several definitions, etc.

And of course, you have pointed out that aria-details can point to any mix of annotations - e.g. a comment and a footnote. In fact, aria-details can point to elements with any role, or mix of roles.

The aria-containsannotations attribute idea sounds interesting. I agree that something like that is better than adding special-purpose roles like commentsection.

Reply:
I don't think we need an actual property aria-containsannotations. If we want, we could compute that infromation automatically for the target of an aria-details, when it is a landmark or generic role.